packages/contracts/contracts/L1/rollup/CanonicalTransactionChain.sol

```diff
@@ -462,39 +447,14 @@ contract CanonicalTransactionChain is ICanonicalTransactionChain, Lib_AddressRes
             curContext = nextContext;
 
             // Process sequencer transactions first.
-            for (uint32 j = 0; j < curContext.numSequencedTransactions; j++) {
-                uint256 txDataLength;
-                assembly {
-                    txDataLength := shr(232, calldataload(nextTransactionPtr))
-                }
-                require(
-                    txDataLength <= MAX_ROLLUP_TX_SIZE,
-                    "Transaction data size exceeds maximum for rollup transaction."
-                );
-
-                leaves[leafIndex] = _getSequencerLeafHash(
-                    curContext,
-                    nextTransactionPtr,
-                    txDataLength,
-                    hashMemory
-                );
-
-                nextTransactionPtr += uint40(TX_DATA_HEADER_SIZE + txDataLength);
-                numSequencerTransactions++;
-                leafIndex++;
-            }
+            numSequencerTransactions += uint32(curContext.numSequencedTransactions);
 
             // Now process any subsequent queue transactions.
-            for (uint32 j = 0; j < curContext.numSubsequentQueueTransactions; j++) {
-                require(
-                    nextQueueIndex < queueLength,
-                    "Not enough queued transactions to append."
-                );
-
-                leaves[leafIndex] = _getQueueLeafHash(nextQueueIndex);
-                nextQueueIndex++;
-                leafIndex++;
-            }
+            nextQueueIndex += uint40(curContext.numSubsequentQueueTransactions);
+            require(
```

Comment:
We could probably save some gas in the average case by moving this check outside of the loop.

Reply:
Nice catch!